pages/docs/tutorials/getting-started/slackbot-websocket-tutorial.md

```diff
@@ -0,0 +1,340 @@
+
+
+Note left of Heart-Counter-Slack-Bot: Event received
+``` 
+
+## Background context
+[WebSocket](https://en.wikipedia.org/wiki/WebSocket) is a communication protocol that enables simultaneous bidirectional data exchange between a client and a server over a single, long-lived connection. Unlike HTTP, which relies on the request-response model, WebSocket is ideal for scenarios where real-time, interactive and low-latency communication is necessary.
+
+
+```mermaid
+sequenceDiagram
+Client -->> Server: Initiate WebSocket Connection
+Server -->> Client: WebSocket Response
+Server -->> Client: WebSocket Response
+Server -->> Client: WebSocket Response
+Server -->> Client: WebSocket Response
+```
+
+In Slack, WebSocket is employed as part of its [Socket Mode](https://api.slack.com/apis/connections/socket) feature to facilitate real-time notifications between Slack's servers and third-party applications or bots. The [Slack Event API](https://api.slack.com/apis/connections/events-api) is a tool that lets you receive real-time notifications of specific events in a Slack workspace such as messages, reactions, and user presence changes.
+
+
+## Define AsyncAPI Version, API Information, and Server
+
+You start your AsyncAPI document by specifying the AsyncAPI version and essential information about your Slack application's API which includes details such as the `title`, `version` and `description`.
+
+The `ws` server section allows you to define the protocol and specify information about the URLs your application will use, such as  `host`, `protocol` and `description`.
+
+<Remember>
+The WebSocket URL is generated  by invoking the <a href="https://api.slack.com/methods/apps.connections.open">apps.connections.open</a> method from Slack’s API. You use the authentication tokens obtained during the configuration of  <code>Heart-Counter-Slack-Bot</code> to generate this URL.
+</Remember>
+
+```
+asyncapi: '3.0.0'
+info:
+  title: Create an AsyncAPI Document for a SlackBot with WebSockets
+  version: '1.0.0'
+  description:  |
+    The Heart-Counter-Slack-Bot manages popular messages in a Slack workspace by monitoring message reaction data using Slack's Event API.
+    
+servers:
+  ws:
+    host: wss://wss-primary.slack.com/
+    protocol: wss
+    description: Websocket URL generated to communicate with Slack
+```
+
+
+## Define Channels and  Bindings
+
+The `channels` attribute defines a communication channel for the event. The `address` specifies where the channel is tuned in to receive messages while the `messages` property defines a key-value pair where each key corresponds to the event it's set up to handle.
+
+The WebSocket URL generated for `Heart-Counter-Slack-Bot` includes authentication tokens, and this protocol-specific data is represented using the `bindings` attribute. By utilizing the `query` object, you can outline the parameters needed for the connection and the conditions they must meet. 
+
+```
+channels:
+  root:
+    address: /
```

Comment:
since we have support for address parameters in v3 I would do it like this:
```
    address: /link/?ticket={ticket}&app_id={app_id}
    parameters:
      ticket:
        default: dfaadfasdfasdfcf
      app_id:
        default: b02112d9364cb2fe
```

now I can specify where these parameters are going to come from , headers or message body. not sure why we still support query parameters for websocket in bindings.

maybe @fmvilas or @derberg can enlighten us here.

Reply:
@KhudaDad414 interesting, indeed text: `Query parameters and fragments SHALL NOT be used, instead use bindings to define them.` is no longer in v3. Dunno though if intentionally or lost in translation 🤔

Reply:
ok, this constraint is still there, query params should be part of the websocket binding

https://github.com/asyncapi/spec/pull/986

Reply:
thanks @derberg. It have become clear now. 🙇